pyro/mesh/patch.py

```diff
@@ -186,6 +189,133 @@ def __eq__(self, other):
         return result
 
 
+class Cartesian2d(Grid2d):
+    """
+    This class defines a 2D Cartesian Grid.
+
+    Define:
+    x = x
+    y = y
+    """
+
+    def __init__(self, nx, ny, ng=1,
+                 xmin=0.0, xmax=1.0, ymin=0.0, ymax=1.0):
+
+        super().__init__(nx, ny, ng, xmin, xmax, ymin, ymax)
+
+        self.coord_type = 0
+
+        # Length of the side in x- and y-direction
+
+        self.Lx = ArrayIndexer(np.full((self.qx, self.qy), self.dx),
+                                 grid=self)
+        self.Ly = ArrayIndexer(np.full((self.qx, self.qy), self.dy),
+                                 grid=self)
+
+        # This is area of the side that is perpendicular to x.
+
+        self.Ax_l = self.Ly.copy()
```

Comment:
oh, I see -- you want it as an array indexer

Reply:
but do we need Lx and Ly ? and if so, do we need to do a copy here?

Reply:
I do Lx and Ly because I think you need them when you construct the normal states for the primitive variable. So for spherical its rdtheta which is not just "dy=dtheta". So I can just pass in Lx and Ly for two different grids.

Reply:
Also I guess you're saying we don't need a copy and just let it be equal since its the same for cartesian

Reply:
yes